core/test/mpi/base/bindings.cpp

```diff
@@ -146,17 +152,52 @@ TYPED_TEST(MpiBindings, CanPutValuesWithLockAll)
     } else {
         data = std::vector<TypeParam>{0, 0, 0, 0};
     }
-    auto win = window(data.data(), 4, comm);
+
+    {
+        auto win = window(data.data(), 4, comm);
+        if (my_rank == 0) {
+            win.lock_all();
+            for (auto rank = 0; rank < num_ranks; ++rank) {
+                if (rank != my_rank) {
+                    win.put(data.data(), 4, rank, 0, 4);
+                }
+            }
+            win.unlock_all();
```

Comment:
You should probably also add tests that actually test the lock. Eg. before this statement, try to access the window on another rank (perhaps in both cases: with and without a separate lock) and test that the expected thing happens. I need to refresh myself on how exactly that's supposed to behave, but it would be nice to have such tests.

Reply:
Yes, I agree that it would be great to have a nice way to test the locks separately. But the problem is that it is difficult to find a consistent and correct way of doing so. The failure is also usually a seg-fault, which makes it difficult to capture it.

Reply:
maybe you can try ASSERT_EXIT like cuda/test/base/cuda_executor_reset if the segmentation fault always return 11 error code. It create another thread to do the test, so I am not sure whether it is suitable for MPI test.

Reply:
Yes, but that is only when an access to the data within the epoch fails. It does not always result in a seg-fault, so you cannot always rely on ASSERT_EXIT.